src/main/java/org/spongepowered/common/util/ReflectionUtil.java

```diff
@@ -77,6 +84,28 @@
         Entity.class
     };
 
+    public static final String REFLECTION_PROPERTY_FILE = "reflection_mappings.json";
+    private static Map<String, String> mappingCache;
+
+    private static String getMethodNameForEnvironment(final String methodName) {
+        if (mappingCache == null) {
```

Comment:
So `Launch` knows if this is a developer environment or not.

I assume that the userdev you have for SpongeFabric exists with official class names and mappings? If so, gate all of this behind the `Launch` `isDeveloperEnvironment` check and just return the mapping. Only production is the issue.

Reply:
This method call is guarded by `isNeighborChangedDeclared` already. Do I move the if-condition from `isNeighborChangedDeclared` to inside this function? It does not make sense for me to guard it twice.

Reply:
Move it in here imo. It does increase the call stack once but does not rely on us remembering that we're already doing that.